### PR TITLE
Add msdf font support

### PR DESCRIPTION
**Pull request by machenmusik**

### CODE REVIEW

File: src/components/text.js [hunk 1]
```diff
@@ -161,7 +163,10 @@ module.exports.Component = registerComponent('text', {
     }
 
     // Shader has changed. Create a shader material.
-    NewShader = createShader(this.el, data.shader, shaderData);
+    // For convenience, if font matches MSDF naming convention, change sdf to msdf.
+    shaderName = data.shader;
+    if (data.shader === 'sdf' && data.font.indexOf('-msdf.') >= 0) { shaderName = 'msdf'; }
```
Comment (ngokevin) [inline src/components/text.js]: A string check is fragile. The shader is configurable, maybe just let people set the shader. But we can switch it if it matches a stock MSDF font.

File: examples/test/text/msdf.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      <a-text mixin="text" font="kelsonsans" position="3 0 0" wrap-count="20"></a-text>
+      <a-text mixin="text" font="kelsonsans" position="3 2 0" wrap-count="30"></a-text>
+      <a-text mixin="text" font="kelsonsans" position="3 3 0" wrap-count="40"></a-text>
+      <a-text mixin="text" font="kelsonsans" position="3 3.5 0" wrap-count="50"></a-text>
+      <a-text mixin="text" font="kelsonsans" position="3 4 0" wrap-count="60"></a-text>
+      <a-text mixin="text" font="kelsonsans" position="3 4.5 0" wrap-count="70"></a-text>
+      <a-text mixin="text" font="kelsonsans" position="3 5 0" wrap-count="80"></a-text>
+
+      <a-sky color="#3CF"></a-sky>
+      <a-entity position="0 2 3.7">
```
Comment (ngokevin) [inline examples/test/text/msdf.html]: In our text examples, I'd try to leave the camera at the default 0, 0, 0, and place the content in the desired places instead.

**Comment by machenmusik** [comment]
hmm, this file is a copy of sizes.html that just uses msdf font instead on the left...

On January 28, 2017 10:26:39 PM EST, Kevin Ngo <notifications@github.com> wrote:
>ngokevin commented on this pull request.
>
>
>
>> +      <a-text mixin="msdf-text" position="-3 3 0"
>wrap-count="40"></a-text>
>+      <a-text mixin="msdf-text" position="-3 3.5 0"
>wrap-count="50"></a-text>
>+      <a-text mixin="msdf-text" position="-3 4 0"
>wrap-count="60"></a-text>
>+      <a-text mixin="msdf-text" position="-3 4.5 0"
>wrap-count="70"></a-text>
>+      <a-text mixin="msdf-text" position="-3 5 0"
>wrap-count="80"></a-text>
>+
>+      <a-text mixin="text" font="kelsonsans" position="3 0 0"
>wrap-count="20"></a-text>
>+      <a-text mixin="text" font="kelsonsans" position="3 2 0"
>wrap-count="30"></a-text>
>+      <a-text mixin="text" font="kelsonsans" position="3 3 0"
>wrap-count="40"></a-text>
>+      <a-text mixin="text" font="kelsonsans" position="3 3.5 0"
>wrap-count="50"></a-text>
>+      <a-text mixin="text" font="kelsonsans" position="3 4 0"
>wrap-count="60"></a-text>
>+      <a-text mixin="text" font="kelsonsans" position="3 4.5 0"
>wrap-count="70"></a-text>
>+      <a-text mixin="text" font="kelsonsans" position="3 5 0"
>wrap-count="80"></a-text>
>+
>+      <a-sky color="#3CF"></a-sky>
>+      <a-entity position="0 2 3.7">
>
>In our text examples, I'd try to leave the camera at the default 0, 0,
>0, and place the content in the desired places instead.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/aframevr/aframe/pull/2325#pullrequestreview-18985568

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by machenmusik** [comment]
it only switches the shader if one is trying to use the default sdf shader with a font that matches msdf pattern (as sdf shader will not work correctly with msdf font image) - if they choose another shader explicitly, this will not apply.

On January 28, 2017 10:25:52 PM EST, Kevin Ngo <notifications@github.com> wrote:
>ngokevin commented on this pull request.
>
>
>
>> @@ -161,7 +163,10 @@ module.exports.Component =
>registerComponent('text', {
>     }
> 
>     // Shader has changed. Create a shader material.
>-    NewShader = createShader(this.el, data.shader, shaderData);
>+    // For convenience, if font matches MSDF naming convention, change
>sdf to msdf.
>+    shaderName = data.shader;
>+    if (data.shader === 'sdf' && data.font.indexOf('-msdf.') >= 0) {
>shaderName = 'msdf'; }
>
>A string check is fragile. The shader is configurable, maybe just let
>people set the shader. But we can switch it if it matches a stock MSDF
>font.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/aframevr/aframe/pull/2325#pullrequestreview-18985558

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by machenmusik** [comment]
(in case it wasn't clear, the default shader change to msdf does apply to the newly added built-in roboto msdf font, but also applies to any others that follow the naming suffix paradigm)

**Comment by machenmusik** [comment]
for yOffset, not sure it's that simple since at the moment, the returned fonts are loaded once and cached for reuse, but having individual yOffset values per font instance would require separate copies of fonts.  thoughts?

**Comment by ngokevin** [comment]
thanks!

**Pull request merged**